src/components/notebook/chem-test.tsx

```diff
@@ -9,23 +11,48 @@ interface IProps {
   chemistryTest: ChemistryTest;
   testIndex: number;
   chemistryTestResults: ChemistryTestResult[];
-  onUpdateTestResult: (type: ChemTestType, completedStep: number) => void;
+  onUpdateTestResult: (type: ChemTestType, completedStep: number, value?: number) => void;
 }
 
 export const ChemTest: React.FC<IProps> = (props) => {
   const { chemistryTest, testIndex, chemistryTestResults, onUpdateTestResult } = props;
   const testResult = chemistryTestResults.find((result) => result.type === chemistryTest.type);
   const stepsComplete = testResult?.stepsComplete ?? 0;
+  const currentStep = stepsComplete > 0
+    ? chemistryTest.steps[Math.min(stepsComplete - 1, chemistryTest.steps.length - 1)]
+    : undefined;
+  const testValueIndex = testResult?.value
+    ? chemistryTest.values.findIndex((val) => val.value === testResult.value)
+    : 0;
+
+  const handleChangeSlider = (event: any, value: number) => {
+    onUpdateTestResult(chemistryTest.type, chemistryTest.steps.length, chemistryTest.values[value].value);
```

Comment:
Not sure whether there's anything to be done about it, but `chemistryTest.values[value].value` isn't the most intuitive access path I've encountered. 😉

Reply:
Yeah, I hate that too.  I changed it to `results`.  That's not quite true and conflicts with some other naming.  But we can refactor later if needed.